src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -169,312 +125,102 @@ public BayesianHetPulldownCalculator(final File refFile, final IntervalList snpI
-        }
-    }
-    /**
-     * Calculate the allele fraction distribution function and its log on the abscissas on the integration
-     * quadrature
-     */
-    private void initializeHetAlleleFractionPrior() {
-
-        alleleFractionPriors.clear();
-        alleleFractionPriors.addAll(gaussIntegrationAbscissas.stream()
-                .mapToDouble(this::calculateAlleleFractionPriorDistribution)
-                .boxed().collect(Collectors.toList()));
-
-        /* calculate the log prior */
-        alleleFractionLogPriors.clear();
-        alleleFractionLogPriors.addAll(alleleFractionPriors.stream()
-                .map(FastMath::log).collect(Collectors.toList()));
+        this.hetPrior = Utils.nonNull(hetPrior);
     }
 
     /**
-     * Calculate the log probability, safeguarded with a minimum probability DEFAULT_MIN_BASE_ERROR_THIRD
-     * @param prob probability
-     * @return safeguarded-log of probability
+     * Calculate the log likelihood of a read pileup is composed of a single nucelotide {@code expectedNucleotide}
+     * @param baseQualities map of bases to list of their calling error probabilities
+     * @param expectedNucleotide the expected nucleotide
+     * @return the log likelihood
      */
-    private double getSafeguardedLogProbability(final double prob) {
-        return FastMath.log(FastMath.max(DEFAULT_MIN_BASE_ERROR_THIRD, prob));
+    private double getSingletonLogLikelihood(final Map<Nucleotide, List<BaseQuality>> baseQualities,
+                                             final Nucleotide expectedNucleotide) {
+
+        return baseQualities.get(expectedNucleotide).stream()
+                .mapToDouble(bq -> GATKProtectedMathUtils.safeLog(1 - bq.getReadErrorProbability() -
+                        3 * bq.getMappingErrorProbability() / 4, MIN_ERROR_PROBABILITY / 3)).sum() +
+                baseQualities.keySet().stream()
+                        .filter(nucl -> nucl != expectedNucleotide)
+                        .map(baseQualities::get).mapToDouble(bqlist -> bqlist.stream()
+                        .mapToDouble(bq -> GATKProtectedMathUtils.safeLog(bq.getReadErrorProbability() / 3 +
+                                bq.getMappingErrorProbability() / 4, MIN_ERROR_PROBABILITY / 3)).sum()).sum();
     }
 
     /**
      * Calculate the log likelihood of a SNP site being homozygous for a given read pileup
      * (see CNV-method.pdf for details)
      * @param baseQualities map of bases to list of their calling error probabilities at the SNP site
-     * @param alleleRef the ref allele base
-     * @param alleleAlt the alt allele base
+     * @param refNucleotide the ref allele nucleotide
+     * @param altNucleotide the alt allele nucleotide
      * @param homRefPrior the prior probability of the ref allele given that the site is homozygous
      * @return the log likelihood
      */
     @VisibleForTesting
     public double getHomLogLikelihood(final Map<Nucleotide, List<BaseQuality>> baseQualities,
```

Comment:
If is visible just for testing, perhaps you can get away with default (package) visibility right? so remove the ```public```.